### PR TITLE
Use EnvUtils to override cache timeout in HTTP ClusterStateProvider

### PR DESCRIPTION
**Pull request by aparnasuresh85**
# Description

The cache timeout to refresh aliases and live nodes in Http ClusterStateProvider is currently hardcoded to 5 seconds. This is pretty frequent.

# Solution

Please provide a short description of the approach taken to implement your solution.

Introducing an override to set the timeout  via EnvUtils that is backwards compatible

# Tests

Please describe the tests you've developed or run to confirm this patch implements the feature or solves the problem.

Ensured existing tests pass
# Checklist

Please review the following and check all that apply:

- [X] I have reviewed the guidelines for [How to Contribute](https://github.com/apache/solr/blob/main/CONTRIBUTING.md) and my code conforms to the standards described there to the best of my ability.
- [ ] I have created a Jira issue and added the issue ID to my pull request title.
- [ ] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [ ] I have developed this patch against the `main` branch.
- [ ] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Reference Guide](https://github.com/apache/solr/tree/main/solr/solr-ref-guide)

### CODE REVIEW

**Comment by aparnasuresh85** [comment]
cc @dsmiley

File: solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java [hunk 1]
```diff
@@ -55,7 +56,8 @@ public abstract class BaseHttpClusterStateProvider implements ClusterStateProvid
   volatile Map<String, Map<String, String>> aliasProperties;
   long aliasesTimestamp = 0;
 
-  private int cacheTimeout = 5; // the liveNodes and aliases cache will be invalidated after 5 secs
+  // the liveNodes and aliases cache will be invalidated after 5 secs
+  private int cacheTimeout = EnvUtils.getPropertyAsInteger("solr.httpcsp.cache.timeout", 5);
```
Comment (janhoy) [inline solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java:60]: Please consider including a category, ref [SIP-21](https://cwiki.apache.org/confluence/display/SOLR/SIP-21%3A+Standardize+system+property+naming). Candiate could be `solr.cloud.httpcsp.cache.timeout`.

Comment (dsmiley) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java:60]: This is specific to SolrJ and people don't know what "httpcsp" is.  Only in the last week did you and I start doing that abbreviation.
`solr.solrj.cache.timeout` and assume milliseconds thus 5000 value (that's the default assumption when units don't clarify)

FYI CloudSolrClient has a cache of DocCollection which has a separate timeout and there's no EnvUtils for that.  So I suppose this is okay to introduce in this PR a very general sounding thing so long as it doesn't mean there might be something more specific with its own timeout.  It's tempting to suggest adding "livenodes" as a name component above but it's both live nodes and aliases, which is awkward.  I'd be okay with that as well, if we think aliases might disappear as an explicit cacheable concern here.

Comment (aparnasuresh85) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java:60]: The timeout was originally specified in seconds, and the elapsed time—used for comparison with the timeout value for livenodes and aliases—is also interpreted in seconds. Therefore, I set the default value to 5 seconds. Let me know if the interpretation and the cachetimeout value needs to be changed to milliseconds.

Comment (dsmiley) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java:60]: Solr has been sloppy in time units but there have been recent improvements on standardization.  So I have no doubts you've seen seconds here & there.  Elsewhere I see a time suffix so it's not ambiguous -- which is ideal.  I recommend adding a ".sec".

Comment (aparnasuresh85) [reply solr/solrj/src/java/org/apache/solr/client/solrj/impl/BaseHttpClusterStateProvider.java]: I updated the env variable name in the latest PR

**Review by dsmiley — APPROVED** [review]
Looks good to me.  Doesn't need a JIRA as it's too simple.  I'll merge this Monday if I don't get more feedback.

**Pull request merged**